optimum/exporters/onnx/__main__.py

```diff
@@ -90,84 +97,93 @@ def main():
+        )
+        onnx_config = onnx_config_constructor(model.config)
 
-    if args.opset < onnx_config.DEFAULT_ONNX_OPSET:
-        raise ValueError(
-            f"Opset {args.opset} is not sufficient to export {model.config.model_type}. "
-            f"At least  {onnx_config.DEFAULT_ONNX_OPSET} is required."
+        needs_pad_token_id = (
+            isinstance(onnx_config, OnnxConfigWithPast)
+            and getattr(model.config, "pad_token_id", None) is None
+            and task in ["sequence_classification"]
         )
-    if args.for_ort and (model.config.is_encoder_decoder or task.startswith("causal-lm")):
-        if model.config.is_encoder_decoder and task.startswith("causal-lm"):
+        if needs_pad_token_id:
+            if args.pad_token_id is not None:
+                model.config.pad_token_id = args.pad_token_id
+            else:
+                try:
+                    tok = AutoTokenizer.from_pretrained(args.model)
+                    model.config.pad_token_id = tok.pad_token_id
+                except Exception:
+                    raise ValueError(
+                        "Could not infer the pad token id, which is needed in this case, please provide it with the --pad_token_id argument"
+                    )
+
+        # Ensure the requested opset is sufficient
+        if args.opset is None:
+            args.opset = onnx_config.DEFAULT_ONNX_OPSET
+
+        if args.opset < onnx_config.DEFAULT_ONNX_OPSET:
             raise ValueError(
-                f"model.config.is_encoder_decoder is True and task is `{task}`, which are incompatible. If the task was auto-inferred, please fill a bug report"
-                f"at https://github.com/huggingface/optimum, if --task was explicitely passed, make sure you selected the right task for the model,"
-                f" referring to `optimum.exporters.tasks.TaskManager`'s `_TASKS_TO_AUTOMODELS`."
+                f"Opset {args.opset} is not sufficient to export {model.config.model_type}. "
+                f"At least  {onnx_config.DEFAULT_ONNX_OPSET} is required."
             )
-        fn_get_models_from_config = (
-            get_encoder_decoder_models_for_export if model.config.is_encoder_decoder else get_decoder_models_for_export
-        )
+        if args.atol is None:
+            args.atol = onnx_config.ATOL_FOR_VALIDATION
+            if isinstance(args.atol, dict):
+                args.atol = args.atol[task.replace("-with-past", "")]
+
+        # Saving the model config and preprocessor as this is needed sometimes.
+        model.config.save_pretrained(args.output.parent)
+        maybe_save_preprocessors(args.model, args.output.parent)
+
+    if task == "stable-diffusion" or (
+        args.for_ort and (model.config.is_encoder_decoder or task.startswith("causal-lm"))
+    ):
+        if task == "stable-diffusion":
+            args.opset = args.opset or 14
+            args.atol = args.atol or 1e-3
```

Comment:
I don't think we should hardcode values here, and we actually set those args to default values above.

Reply:
This means that the default ospet and atol of the onnx config will not be used when not specified by the user. I would rather the make the opset argument optional in `export `, and use the onnx config default value when not specified

Reply:
But already we check if `args.opset` is `None` and set it to the onnx config value if that's the case. Same thing for the `atol`.

Reply:
Yes, this is done if the model has an `OnnxConfig`, but in our case each components has it's own (so we don't have a default `opset` or `atol` to give to `export_models` as every components should have his own default value). I will make the `opset` / `atol` optional in both `export` and `validate_model_outputs`